FireboltNETSDK/Client/FireboltConnection.cs

```diff
@@ -52,6 +54,8 @@ public FireboltClient Client
         private FireboltClient? _fireboltClient;
         public readonly HashSet<string> SetParamList = new();
         private int _infraVersion = 0;
+        private static IDictionary<string, GetAccountIdByNameResponse> accountCache = new ConcurrentDictionary<string, GetAccountIdByNameResponse>();
+        private ISet<CacheListener> listeners = new HashSet<CacheListener>();
```

Comment:
Why do we add listener support here? Any scenario where it would be useful to register internal cache cleanup events?

Reply:
This is one of the OOD principles: decoupling.

Reply:
Please elaborate on this. Otherwise OOD principle: overengineering comes to mind.

>  Any scenario where it would be useful to register internal cache cleanup events?

Reply:
I cannot elaborate because I do not understand what's the problem?
I have added cache for account data. I have also added method Cleanup() that clears the cache stored in the connection level. But there is data cached on FireboltClient level. This cache should be cleaned as well. So, I added the following logic. Firebolt client "registers" itself calling method Connection.Regiser(this) (seem FireboltClient's constructor). The method Register adds the client into the list, so than, when Connection.Cleanup() is called it can iterate over this list and clean the clients also. 

Could I store collection of FireboltClients? Yes, surely. But I preferred to define interface CacheListener that so far has only one method (although it can be expanded in future). So, Connection "does not know" that it cleans up FireboltClients. It cleans up cache listeners. This is the decoupling that I mentioned above.

Reply:
> I cannot elaborate because I do not understand what's the problem?

There is no problem. I was asking you to clarify the use case for it. This comment is a _question_.
As per my (limited) understanding we were doing a cleanup anyway so I did not see where the listener was used. All I needed is an explanation

Reply:
Anyway this is irrelevant now because this code is removed after discussion with Stepan